wofs/virtualproduct.py

```diff
@@ -34,18 +37,39 @@ def measurements(self, input_measurements) -> Dict[str, Measurement]:
         return self.output_measurements
 
     def compute(self, data) -> Dataset:
+
         print(data.geobox)
         print(repr(data.geobox))
+
+        masking_used = fmask_filter
+        if self.c2:
+            masking_used = fmask_filter_c2
+            # The C2 data needs to be scaled
+            bands = ['nbart_blue', 'nbart_green', 'nbart_red', 'nbart_nir', 'nbart_swir_1', 'nbart_swir_2']
+
+            for band in bands:
+                dtype = data[band].dtype
+                nodata = data[band].attrs['nodata']
+                attrs = data[band].attrs
+
+                is_valid_array = valid_data_mask(data[band])
+
+                data[band] = data[band].where(is_valid_array)
```

Comment:
note: `float64` is probably not needed here? Or is input already `float32`  and not `int16`.

`.where()` up-casts to `float64` for integer types.